### DISCUSSION TITLE
Superbase db query from URL

### DISCUSSION THREAD

**Discussion by nickelliott71** [supabase/supabase#12028] [Questions]
I'm struggling to formulate a supabase db query from multiple tables, whilst using a value from a URL.

I have 3 tables (simplified)...

```
series > id, series
authors > id, name
books > id, title, series_id(fk), authors_id(fk)
```

I want to get a list of books in a series, along with the author and series name from the following URL...

app.com/series. i.e. app.com/harrypotter

I can get the series name using getServerSideProps, but struggling how to write my query to supabase. Every way I write it either gives me a NULL object or a 500 error.

I feel like I should be querying the books table and then be able to get the series and author names through the foreign keys. But the query that it's centred around is the series name, which is in the series table. So unsure of the db query to do it, or whether I should structure my db table's in a different way?

```
  export async function getServerSideProps( context ) {
      const { series } = context.query;
        ...
      return {
        props: {
        ...
        }
      };
    }
```

Thanks for any help in advance (from someone who hasn't coded for 10 years)!!

**Comment by steve-chavez** [comment]
Hello,

> Every way I write it either gives me a NULL object or a 500 error.

What's the error message in that case? Also, can you share the JS snippet you're trying?
